src/neo-vm/Types/Array.cs

```diff
@@ -78,12 +80,14 @@ public Array(ReferenceCounter? referenceCounter, IEnumerable<StackItem>? items =
         /// <param name="item">The item to be added.</param>
         public void Add(StackItem item)
         {
+            if (IsReadOnly) throw new InvalidOperationException();
             _array.Add(item);
             ReferenceCounter?.AddReference(item, this);
         }
 
         public override void Clear()
         {
+            if (IsReadOnly) throw new InvalidOperationException();
```

Comment:
I prefer to use `ReadOnlyException(this)` to make it easier to debug.

Reply:
Done.